### DISCUSSION TITLE
Will it be possible to add any 3060 12GB links before its release tomorrow 2/25?

### DISCUSSION THREAD

**Discussion by l00sed** [jef/streetmerchant#2024] [Ideas and feature requests]
I learned that you are able to add a store through the instructions found in this document:
https://jef.codes/streetmerchant/help/general/#adding-a-store
I'm wondering if new stores or products are found and then added, or if there is any component in streetmerchant that allows for scraping to discover new product URLs? Specifically, I'm wondering if there are any URLs of 3060 12GB models that exist yet.

**Comment by damiankrzyz** [comment]
I tried to add it manually yesterday but was unsuccessful. Can anyone help?

**Reply by damiankrzyz** [reply]
In the UK Currys and eBuyer has already got the links:
https://www.currys.co.uk/gbuk/rtx-3060/components-upgrades/graphics-cards/324_3091_30343_xx_ba00013562-bv00314002/xx-criteria.html
https://www.ebuyer.com/store/Components/cat/Graphics-Cards-Nvidia/subcat/GeForce-RTX-3060

**Reply by l00sed** [reply]
I think everything you need to add that is in the documentation that I linked to above. Although, it looked like we'll have to wait until an "order" or "add to cart" button becomes visible on the page to add in the automation. This kind of defeats the purpose though, I guess, if you're already one of the first to see the product drop and it is readily available for purchase...

**Reply by damiankrzyz** [reply]
I don't think we have to wait until then. At least not for some shops. I managed to get it working now for 3060 12GB.
I had to also modify the discord notification config that wasn't present in the documentation. 
![Screenshot 2021-02-25 at 11 38 40](https://user-images.githubusercontent.com/79311645/109148530-4c8ef300-775e-11eb-8660-8c80c9bb41e0.png)

**Reply by l00sed** [reply]
I've collected some links for US stores, only a few; but it's still a dozen SKUs.
As soon as I insert `3060` in `store.ts`, I can't compile:

## store.ts
![image](https://user-images.githubusercontent.com/23065167/109169105-7a1f7080-774d-11eb-8474-044f70b2ac99.png)

3000+ errors with every store.... `The expected type comes from property 'series' which is declared here on type 'Link'`
![image](https://user-images.githubusercontent.com/23065167/109169204-91f6f480-774d-11eb-91a2-658afc31ee14.png)

The full error log is quite long, but more of the same.

**Comment by l00sed** [comment]
I'm just swapping out links for 3060ti models with 3060 12GB links as I'm finding them.

**Reply by l00sed** [reply]
As a work-around

**Reply by damiankrzyz** [reply]
also need to update the config file and then for each individual shop
![Screenshot 2021-02-25 at 15 33 58](https://user-images.githubusercontent.com/79311645/109176727-01d1a300-777f-11eb-9345-58cde6f0ed31.png)

**Reply by l00sed** [reply]
# config.ts
![image](https://user-images.githubusercontent.com/23065167/109181259-b8229180-7759-11eb-8128-80c177023ce3.png)
# store.ts
![image](https://user-images.githubusercontent.com/23065167/109181376-d9837d80-7759-11eb-9717-8f79baf88dd2.png)

# npm run start
![image](https://user-images.githubusercontent.com/23065167/109181520-00da4a80-775a-11eb-8025-dd25ca4d7ee8.png)

**Reply by damiankrzyz** [reply]
did you try "npm install"?

**Reply by l00sed** [reply]
Would I have to run that after the initial npm install?

**Reply by l00sed** [reply]
Trying that out now. Thanks for the suggestion.

**Reply by l00sed** [reply]
Re-ran npm install.
Same issue. As soon as I add '3060' to src/store/model/store.ts it blows up. Compiles and runs fine without it.
![image](https://user-images.githubusercontent.com/23065167/109185220-cd012400-775d-11eb-99ce-05fda06346a1.png)


This was really the only info in the logs:
```
9 verbose lifecycle streetmerchant@~start: CWD: /home/dan/Documents/streetmerchant
10 silly lifecycle streetmerchant@~start: Args: [ '-c', 'npm run compile && node build/src/index.js' ]
11 silly lifecycle streetmerchant@~start: Returned: code: 1  signal: null
12 info lifecycle streetmerchant@~start: Failed to exec start script
13 verbose stack Error: streetmerchant@ start: `npm run compile && node build/src/index.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/home/dan/.nvm/versions/node/v10.21.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:198:13)
13 verbose stack     at ChildProcess.<anonymous> (/home/dan/.nvm/versions/node/v10.21.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:198:13)
13 verbose stack     at maybeClose (internal/child_process.js:982:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid streetmerchant@
15 verbose cwd /home/dan/Documents/streetmerchant
16 verbose Linux 5.4.0-66-generic
17 verbose argv "/home/dan/.nvm/versions/node/v10.21.0/bin/node" "/home/dan/.nvm/versions/node/v10.21.0/bin/npm" "run" "start"
18 verbose node v10.21.0
19 verbose npm  v6.14.4
20 error code ELIFECYCLE
21 error errno 1
22 error streetmerchant@ start: `npm run compile && node build/src/index.js`
22 error Exit status 1
23 error Failed at the streetmerchant@ start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```
I'm not sure how helpful this is to me...

**Comment by machineica** [comment]
can anyone share how to do this for the 3060? this has to be done in .ts? or can the dotenv be edited for it?

Have my eye on the best buy link 
https://www.bestbuy.com/site/evga-nvidia-geforce-rtx-3060-xc-gaming-12gb-gddr6-pci-express-4-0-graphics-card/6454328.p?skuId=6454328

**Reply by l00sed** [reply]
There are a few bestbuy and newegg links that can be adjusted. I know you'll have to edit the .ts files add models, this can't be configured solely in the DOTENV file.
